eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -97,14 +98,15 @@ case class Commitments(channelId: ByteVector32,
    * @param scriptPubKey optional local script pubkey provided in CMD_CLOSE
    * @return the actual local shutdown script that we should use
    */
-  def getLocalShutdownScript(scriptPubKey: Option[ByteVector]): Either[ChannelException, ByteVector] = {
+  def getLocalShutdownScript(scriptPubKey: Option[ByteVector], defaultScriptPubKey: ByteVector): Either[ChannelException, ByteVector] = {
     // to check whether shutdown_any_segwit is active we check features in local and remote parameters, which are negotiated each time we connect to our peer.
     val allowAnySegwit = Features.canUseFeature(localParams.initFeatures, remoteParams.initFeatures, Features.ShutdownAnySegwit)
     (channelFeatures.hasFeature(Features.UpfrontShutdownScript), scriptPubKey) match {
-      case (true, Some(script)) if script != localParams.defaultFinalScriptPubKey => Left(InvalidFinalScript(channelId))
+      case (true, Some(script)) if script != localParams.defaultFinalScriptPubKey.get => Left(InvalidFinalScript(channelId))
```

Comment:
I would simplify by matching on the script:
```scala
(upfrontShutdownScript_opt, requestedScriptPubKey_opt) match {
  ...
}
```

Reply:
yes it's better like this, done in https://github.com/ACINQ/eclair/pull/2565/commits/21a46ca164a11a5954b87d4569fc77b30954ef0a